dbt_meshify/utilities/references.py

```diff
@@ -142,11 +144,28 @@ def update_child_refs(self, resource: CompiledNode) -> ChangeSet:
             if not hasattr(model_node, "language") or not isinstance(model_node, CompiledNode):
                 continue
 
+            if current_change_set:
+                previous_changes: List[FileChange] = [
+                    change
+                    for change in current_change_set.changes
+                    if (
+                        isinstance(change, FileChange)
+                        and change.identifier == model_node.name
+                        and change.operation == Operation.Update
+                        and change.entity_type == EntityType.Code
+                        and change.path
+                        == self.project.parent_project.resolve_file_path(model_node)
+                    )
+                ]
+                previous_change = previous_changes[-1] if previous_changes else None
+
             change = self.generate_reference_update(
                 project_name=self.project.name,
                 upstream_node=resource,
                 downstream_node=model_node,
-                code=model_node.raw_code,
+                code=previous_change.data
+                if (previous_change and previous_change.data)
+                else model_node.raw_code,
```

Comment:
This approach seems reasonable. Since this filtering is being done here and in `linker.py`, I think it might be appropriate to extract this logic into a function in this file (`references.py`) that both this method and the `linker.py` method can leverage.

Reply:
totally agree - would this be a method on the `ReferenceUpdater` Class, or just a function outside the scope of the class?

Reply:
I'm thinking outside the `ReferenceUpdater` class scope.